src/release/upgrade.md

```diff
@@ -19,22 +19,67 @@ $ flutter upgrade
 This command gets the most recent version of the Flutter SDK
 that's available on your current Flutter channel.
 
-If you want an even more recent version of the Flutter SDK,
-switch to a less stable Flutter channel
+If you are using the **stable** channel
+and want an even more recent version of the Flutter SDK,
+switch to the **beta** channel using `flutter channel beta`,
 and then run `flutter upgrade`.
 
+### Keeping informed
+
+We publish [migration guides] for known breaking changes.
+
+We send announcements regarding these changes to the
+[Flutter announcements mailing list][flutter-announce].
+
+To avoid being broken by future versions of Flutter,
+consider submitting your tests to our [test registry].
+
 
 ## Switching Flutter channels
 
-Flutter has three [release channels][]:
-**stable**, **beta** and **master**.
+Flutter has two release channels:
+**stable** and **beta**.
 
-{{site.alert.info}}
-  The `dev` channel was retired as of Flutter 2.8.
-{{site.alert.end}}
+### The **stable** channel
+
+We recommend the **stable** channel
+for new users
+and for production app releases.
+It is updated roughly every three months,
+and may additionally receive occasional hot fixes
+for high-severity or high-impact issues.
```

Comment:
**nits:** Write in active voice per [GSG](https://developers.google.com/style/voice). Limit adverbs as they make Google sound unsure.
```suggestion
We recommend the **stable** channel for new users
and for production app releases.
The team updates this channel about every three months.
The channel may receive occasional hot fixes
for high-severity or high-impact issues.
```